Sources/NIOHTTP1/HTTPUpgradeHandler.swift

```diff
@@ -187,7 +187,7 @@ public class HTTPServerUpgradeHandler: ChannelInboundHandler {
         let allHeaderNames = Set(request.headers.map { $0.name.lowercased() })
 
         for proto in requestedProtocols {
-            guard let upgrader = upgraders[proto] else {
+            guard let upgrader = upgraders[proto.lowercased()] else {
```

Comment:
This should be accompanied by an equivalent lowercasing when we create the `upgraders` dictionary.

Reply:
@Lukasa Thanks! Will add the tests :)

Reply:
@ianpartridge Not really.

The spec is silent on whether or not case is meaningful in the upgrade header values, but I'd want to take my cue from the Connection header values, which end up being treated case insensitively for maximum compatibility. I'd suspect the same is true here, and I sincerely doubt the IETF would ever produce a specification which allowed two protocols to register an upgrade token distinguished only by case.